lib/rubocop/options.rb

```diff
@@ -57,6 +57,9 @@ def validate_compatibility
          (@options[:only] & %w(Lint/UnneededDisable UnneededDisable)).any?
         fail ArgumentError, 'Lint/UnneededDisable can not be used with --only.'
       end
+      if @options.key?(:cache) && @options.key?(:auto_correct)
+        fail ArgumentError, '--cache can not be used with --auto-correct.'
```

Comment:
Yes, probably. Either way, it makes no sense to have this check as caching could also be turned on in configuration.